Automattic/wp-calypso
# Importers: Move file upload to redux

#### Changes proposed in this Pull Request

This PR is an attempt to break out some of the work found in #27425 in to a more reviewable chunk.
It aims to move the importer upload state from the flow store in to our redux store.

To achieve this, I've had to make a slight compromise in how this works but the effect should still be the same as before...
Previously, when somebody started an upload, the details for that upload would be held against a fake importer id (see #27425 for details on this fake id/object). That object would later be replaced when the upload completes, along with the upload data.
These changes now assume just a single upload at any one time as we hope to get rid of the need for the fake id/object.

#### Testing instructions


#### Testing the regular flow

- Start at http://calypso.localhost:3000/settings/import
- Go to the WordPress importer by clicking "Start Import"
- Upload a valid WXR file
  - If slow enough, you should see a progress bar - often the exported file is small, so you might miss this
  - If not slow enough, you'll see the progress later when we attempt to upload a large file.
  - You should see the UI progress to the map authors pane
  - After mapping authors, you should progress through the import and eventually get to the 'import complete' state
- Repeat this process for the other importers
  - Note that site importer is a little different given the url and preview steps 

#### Testing errors
- Go to the WordPress importer by clicking "Start Import"
- Upload a non WXR file, this is a good opportunity to see the progress bar, by choosing a large file - Note, we might want to consider checking for file extension to help avoid people dropping invalid files here earlier
  - Once uploaded, you should see an error as below:

<img width="654" alt="screen shot 2018-10-12 at 09 38 52" src="https://user-images.githubusercontent.com/4335450/46858205-b531db00-ce02-11e8-9775-8acbbf14d390.png">
